### PR TITLE
[Improvement-14354][Parameter] Rename Property to Parameter

### LINKED ISSUE #14354: [Improvement][Parameter] Rename Property to Parameter
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar feature requirement.

### Description

Rename Property to Parameter

### Are you willing to submit a PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by rickchengx**
## Purpose of the pull request

* close: #14354 

## Brief change log

* Rename `Property` to `Parameter`
* Rename `prop` to `key`

## Verify this pull request

截屏2023-06-15 11 55 08

截屏2023-06-15 10 37 34

### CODE REVIEW

**Comment by Radeity** [comment]
Hi, can we modify parameter name, either?

**Comment by rickchengx** [comment]
> Hi, can we modify parameter name, either?

@Radeity Sure, you mean sth like this?

image

**Comment by Radeity** [comment]
> > Hi, can we modify parameter name, either?
> 
> @Radeity Sure, you mean sth like this?
> 
> image

Yes, that's clear now, thanks for your effort :D

**Review by Radeity — APPROVED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/model/Parameter.java [hunk 1]
```diff
@@ -23,13 +23,13 @@
...
 
-public class Property implements Serializable {
+public class Parameter implements Serializable {
 
     private static final long serialVersionUID = -4045513703397452451L;
     /**
      * key
      */
-    private String prop;
+    private String key;
```
Comment (ruanwenjun) [inline dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/model/Parameter.java:32]: It seems this pr will change the metadata structure, we need to refresh the whole data in db.

**Review by ruanwenjun — CHANGES_REQUESTED** [review]

File: dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/model/Parameter.java (hunk 1 above)
Comment (rickchengx) [reply dolphinscheduler-task-plugin/dolphinscheduler-task-api/src/main/java/org/apache/dolphinscheduler/plugin/task/api/model/Parameter.java:32]: @ruanwenjun I got you. So it seems that `prop` should remain unchanged, and only rename `Property` to `Parameter`, WDYT?

**Review by songjianet — APPROVED** [review]

**Pull request closed without merging**